src/main/kotlin/org/rust/lang/core/resolve/Processors.kt

```diff
@@ -511,3 +514,41 @@ fun filterDeriveProcMacros(processor: RsResolveProcessor): RsResolveProcessor =
         if (!function.isCustomDeriveProcMacroDef) return@wrapWithFilter false
         true
     }
+
+fun RsOuterAttributeOwner.shouldHideElementInCompletion(contextMod: RsMod): Boolean {
+    val elementContainingMod = containingMod
+    if (elementContainingMod == contextMod) return false
+
+    // Hide `#[doc(hidden)]` items
+    if (queryAttributes.isDocHidden) return true
+
+    val rustcChannel = contextMod.cargoProject?.rustcInfo?.version?.channel ?: return false
+    val showUnstableItems = rustcChannel != RustChannel.STABLE && rustcChannel != RustChannel.BETA
+    if (showUnstableItems) return false
+    if (containingCrate.origin != PackageOrigin.STDLIB) return false
+
+    // Hide unstable stdlib items. The absence of `#[stable]` attribute in stdlib means the item is unstable
+    return stability != RsStability.Stable
+}
+
+private val RsOuterAttributeOwner.stability: RsStability?
+    get() {
+        // Own stability
+        queryAttributes.stability?.let { return it }
+
+        if (this is RsAbstractable) {
+            val owner = owner
+            if (owner is RsAbstractableOwner.Impl) {
+                val ownerStability = owner.impl.queryAttributes.stability
+                if (ownerStability != RsStability.Stable) {
+                    return ownerStability
+                }
+                val superItem = superItem
+                if (superItem != null) {
+                    return superItem.queryAttributes.stability
+                }
+            }
+        }
+
+        return null
```

Comment:
Looks like modules by default are `#[stable]`, should we handle them separately here?

```rust
// stdlib
pub mod inner {
    #[stable]
    pub fn func() {}
}

// main.rs
fn main() {
    // currently no completion for inner
    inn/*caret*/::func();
}
```